Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,189 @@
+
+    /// <summary>
+    /// Diagnostic listener implementation that listens for events specific to outgoing depedency requests.
+    /// </summary>
+    public class DependencyCollectorDiagnosticListener : IObserver<DiagnosticListener>
+    {
+        /// <summary>
+        /// Add Application Insights Dependency Collector services to this .NET Core application.
+        /// </summary>
+        /// <returns>
+        /// An IDisposable that can be disposed to disable the DependencyCollectorDiagnosticListener.
+        /// </returns>
+        public static IDisposable Enable(TelemetryConfiguration configuration = null)
+        {
+            if (configuration == null)
+            {
+                configuration = TelemetryConfiguration.Active;
+            }
+
+            return DiagnosticListener.AllListeners.Subscribe(new DependencyCollectorDiagnosticListener(configuration));
+        }
+
+        /// <summary>
+        /// Source instrumentation header that is added by an application while making http requests and retrieved by the other application when processing incoming requests.
+        /// </summary>
+        internal const string SourceInstrumentationKeyHeader = "x-ms-request-source-ikey";
+
+        /// <summary>
+        /// Target instrumentation header that is added to the response and retrieved by the calling application when processing incoming responses.
+        /// </summary>
+        internal const string TargetInstrumentationKeyHeader = "x-ms-request-target-ikey";
+
+        /// <summary>
+        /// Standard parent Id header.
+        /// </summary>
+        internal const string StandardParentIdHeader = "x-ms-request-id";
+
+        /// <summary>
+        /// Standard root id header.
+        /// </summary>
+        internal const string StandardRootIdHeader = "x-ms-request-root-id";
+
+        private readonly ApplicationInsightsUrlFilter applicationInsightsUrlFilter;
+        private readonly TelemetryClient client;
+        private readonly ConcurrentDictionary<Guid, DependencyTelemetry> pendingTelemetry = new ConcurrentDictionary<Guid, DependencyTelemetry>();
+
+        internal DependencyCollectorDiagnosticListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+
+            // Since dependencySource is no longer set, sdk version is prepended with information which can identify whether RDD was collected by profiler/framework
+            // For directly using TrackDependency(), version will be simply what is set by core
+            string versionStr = typeof(DependencyCollectorDiagnosticListener).GetTypeInfo().Assembly.GetCustomAttributes()
+                    .OfType<AssemblyFileVersionAttribute>()
+                    .First()
+                    .Version;
+
+            Version version = new Version(versionStr);
+            this.client.Context.GetInternalContext().SdkVersion = "rddf" + version.ToString(3) + "-" + version.Revision;
```

Comment:
I learned that you can provide build constants in the project.json file for conditional compilation. I've defined NETCORE and then added conditions in the code to use .NET Core specific code in certain places. That helped me to not duplicate other functionality elsewhere (InstrumentationKeyHashLookupHelper, for example).